hack/api-reference/cilium.md

```diff
@@ -430,6 +444,82 @@ bool
 </tr>
 </tbody>
 </table>
+<h3 id="cilium.networking.extensions.gardener.cloud/v1alpha1.Encryption">Encryption
+</h3>
+<p>
+(<em>Appears on:</em>
+<a href="#cilium.networking.extensions.gardener.cloud/v1alpha1.NetworkConfig">NetworkConfig</a>)
+</p>
+<p>
+<p>Encryption enables transparent network encryption for cilium</p>
+</p>
+<table>
+<thead>
+<tr>
+<th>Field</th>
+<th>Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>
+<code>enabled</code></br>
+<em>
+bool
+</em>
+</td>
+<td>
+</td>
+</tr>
+<tr>
+<td>
+<code>keyFile</code></br>
+<em>
+string
+</em>
+</td>
+<td>
+<em>(Optional)</em>
+<p>KeyFile is the name of the key file inside the Kubernetes secret configured via SecretName</p>
+</td>
+</tr>
+<tr>
+<td>
+<code>mountPath</code></br>
+<em>
+string
+</em>
+</td>
+<td>
+<em>(Optional)</em>
+<p>MountPath is the path where to mount the secret inside the Cilium pod</p>
+</td>
+</tr>
+<tr>
+<td>
+<code>secretName</code></br>
+<em>
+string
+</em>
+</td>
+<td>
+<p>SecretName is the name of the Kubernetes secret containing the encryption</p>
```

Comment:
Who is creating the secret?
Is that something that could be done in the extension?

Reply:
Best user experience is to create the secret in the extension.

Reply:
@axel7born secret management is now also part of this PR. please review